### DISCUSSION TITLE
MSVC Version is older than claimed

### DISCUSSION THREAD

**Discussion by danakj** [community/community#77793] [Enterprise]
### Select Topic Area

Bug

#### Claim from docs

https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#visual-studio-enterprise-2022 claims that the Windows 2022 runner should have Visual Studio 2022 version 17.8.34316.72. While 17.8.34316.72 is a _build number_, it [corresponds to](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-history) Visual Studio 2022 **version 17.8.1**.

#### Actual observed version

When running cmake configure on a Windows machine, it finds `C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.37.32822/bin/HostX64/x64/cl.exe`

The MSVC version 14.37.32822 [corresponds to](https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B) Visual Studio 2022 **version 17.7.0**

#### Expected version based on docs

Whereas, I have Visual Studio 2022 version 17.8.1 installed locally and it comes with MSVC 14.38.33130, at `C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.38.33130/bin/Hostx86/x64/cl.exe`, which is actually from build 17.8.34316.72.

I don't think I am holding this wrong, cmake is finding what it finds, but maybe?

#### Links

Example CI run: https://github.com/chromium/subspace/actions/runs/7078666426/job/19264399269?pr=415
Debugging happened here so far: https://github.com/chromium/subspace/issues/267

**Comment by dakshithadissanayaka** [comment]
Sure, here is a summary of the discussion and a possible solution:

**Summary:**

The user is experiencing an issue where the Windows 2022 runner is using an older version of MSVC (14.37.32822) than the version that is documented (14.38.33130). This is causing problems when running CMake, as it is finding the older version of MSVC and using that instead of the newer version.

**Possible Solution:**

There are a few possible solutions to this issue:

1. **Update the documentation:** The documentation could be updated to reflect the actual version of MSVC that is being used by the Windows 2022 runner. This would help to prevent other users from encountering the same problem.
2. **Use a specific version of MSVC:** The user could specify a specific version of MSVC to use when running their CI jobs. This would ensure that the correct version of MSVC is used, even if the documentation is not updated.
3. **Update the runner:** The runner could be updated to use the newer version of MSVC. This would be the most permanent solution, as it would fix the problem for all users.

**Additional Notes:**

* The user has already tried updating their local version of Visual Studio to 17.8.1, but this did not resolve the issue. This suggests that the problem is with the runner, not with the user's local environment.
* The user has also tried using a specific version of MSVC, but this did not work either. This suggests that the runner is not properly respecting the user's specified version of MSVC.
* It is possible that the runner is configured to use the older version of MSVC for compatibility reasons. However, this is not documented and it is causing problems for some users.

I hope this summary is helpful. Please let me know if you have any other questions.
